app/services/qa/linked_data/response_header_service.rb

```diff
@@ -0,0 +1,78 @@
+# Service to construct a request header that includes optional attributes for search and fetch requests.
+module Qa
+  module LinkedData
+    class ResponseHeaderService
+      class_attribute :ldpath_service
+      self.ldpath_service = Qa::LinkedData::LdpathService
+
+      attr_reader :request_header, :results, :config, :graph
+
+      # @param request_header [Hash] request attributes constructed by Qa::LinkedData::RequestHeaderService
+      # @param results [String, JSON] results generated by executing a request
+      # @param config [String] authority configuration identifying where in the results or request_header data can be found
+      # @param graph [RDF::Graph] graph holding results
+      def initialize(request_header:, results:, config:, graph:)
+        @request_header = request_header
+        @results = results
+        @config = config
+        @graph = graph
+      end
+
+      # Construct response header to pass back with search results (linked data module).
+      # @returns [Hash] response data
+      # @see Qa::Authorities::LinkedData::SearchQuery
+      def search_header
+        header = {}
+        header[:start_record] = start_record
+        header[:requested_records] = requested_records
+        header[:retrieved_records] = retrieved_records
+        header[:total_records] = total_record_count
+        header
+      end
+
+      # Construct response header to pass back with fetch results (linked data module).
+      # @returns [Hash] response data
+      # @see Qa::Authorities::LinkedData::FetchTerm
+      def fetch_header
+        header = {}
+        header[:predicate_count] = pred_count
+        header
+      end
+
+      private
+
+        # determine the position of the first record in the returned results relative to all search results
+        def start_record
+          request_header.fetch(:replacements, {}).fetch(config.start_record_parameter, 1).to_i
```

Comment:
The actual value of the start_record is passed in by the caller in a parameter.

`config.start_record_parameter` identifies the name of the parameter that is used to pass in startRecord value.

`request_header[:replacements]` holds parameters passed into QA that are forwarded through to the external authority request.  It has a hash key that is the name of the start_record_parameter and is used to get the value from the request_header